### ISSUE TITLE
issue #582 does not solve the problem with multiple VPC endpoints matched on Terraform 0.12 and aws provider 2.70

### ISSUE THREAD

**Issue by dgonzalez** [terraform-aws-modules/terraform-aws-vpc#585]
In the issue it is recommended for people who cannot update the aws provider to version 3.10 to use the release 2.69.0 of terraform-aws-vpc.

I am in such situation (upgrading the provider is not trivial, we are working on it but will take us days if not a couple of weeks) and when I try to plan with the 2.69.0 I get the following error:
```
Error: "service_type": this field cannot be set

  on .terraform/modules/this.vpc/vpc-endpoints.tf line 4, in data "aws_vpc_endpoint_service" "s3":
   4: data "aws_vpc_endpoint_service" "s3" {



Error: "service_type": this field cannot be set

  on .terraform/modules/this.vpc/vpc-endpoints.tf line 45, in data "aws_vpc_endpoint_service" "dynamodb":
  45: data "aws_vpc_endpoint_service" "dynamodb" {
```

This is using the AWS provider 2.70.0. Am I wrong or it is not possible to use this module with a provider below 3.10 after AWS released the S3 PrivateLink interface?

**Comment by circa10a** [comment]
@dgonzalez I have the same problem, here's what I did to get around it until we can upgrade the provider: https://github.com/hashicorp/terraform-provider-aws/issues/17417

**Comment by tbugfinder** [comment]
Hi @circa10a ,
as you have already fixed it, could open open a PR to this repo?
IMHO, fixes for the issue within this repo were applied too quickly as there are means to serve all provider versions.

It would be excellent getting some feedback from maintainers.

**Comment by bryantbiggs** [comment]
hi all, could you try out the changes proposed here https://github.com/terraform-aws-modules/terraform-aws-vpc/pull/597 to see if this resolves the issues you are currently facing please?

**Comment by tbugfinder** [comment]
Hi @bryantbiggs ,
your PR still requires provider v3.10.
We are looking for a fix with provider v2.70.0 compatibility.

https://github.com/bryantbiggs/terraform-aws-vpc/blob/fix/s3-interface-endpoint/versions.tf#L7

**Comment by bryantbiggs** [comment]
ah, you are right @tbugfinder - I forgot that change. I've just pushed that change up after testing/validating
